### PR TITLE
[RFC][Formatting] Add scripts for applying Black to the Python code.

### PR DESCRIPTION
**Pull request by jroesch**
As per the recent RFC https://discuss.tvm.apache.org/t/rfc-introduce-automatic-formatting-of-python-code/7843/10.

### CODE REVIEW

**Comment by jroesch** [comment]
cc @tqchen @areusch @comaniac @u99127 @jwfromm @mbrookhart @junrushao1994

**Comment by junrushao1994** [comment]
It is a bit hard to review 1000 files...maybe just take a look at the pyproject.toml file and assume other parts are correct?

File: tests/lint/git-black.sh
Comment (comaniac) [inline tests/lint/git-black.sh:32]:
```suggestion
    echo "Run black-format on Python files that changed since <commit>"
```

File: tests/lint/git-black.sh
Comment (comaniac) [inline tests/lint/git-black.sh]:
```suggestion
echo "Running git-black on Python files against" $1
```

File: pyproject.toml
Comment (comaniac) [inline pyproject.toml:19]: So we are using 88 as the TVM standard column length for Python codes? I know this is the default and the suggested value in black so just want to double confirm.

**Review by comaniac — CHANGES_REQUESTED** [review]
Tried to review the first 3 commits because I cannot even open the changes after the formatting commit...

I'd suggest separating two 2 PRs. The first one focuses on the scripts and CI flow (but disable the checking to pass the CI). The second one then formats all Python codes. In this way, as long as the second PR passes the CI we should be be good.

File: tests/lint/git-black.sh
Comment (areusch) [inline tests/lint/git-black.sh:57]: can we use array style here?

`IFS=$'\n' read -a FILES -d'\n' < <(git diff --name-only HEAD $1 -- "*.py" "*.pyi")`
`CMD=( "black" "--check" "${FILES[@]}" )`
`"${CMD[@]}"`

**Review by areusch — CHANGES_REQUESTED** [review]
+1 to splitting the review into a scripts part and a format part.

**Comment by jroesch** [comment]
@areusch @tqchen @comaniac I can rollback the formatting, the first 3 or 4 commits were focused on formatting then I went through the process to see if it would actually work.

File: tests/lint/git-black.sh
Comment (jroesch) [reply tests/lint/git-black.sh:57]: honestly I hate bash and struggle through as much as possible will defer to you

File: pyproject.toml
Comment (jroesch) [reply pyproject.toml:19]: @tqchen confirmed this should be 100.

**Comment by jroesch** [comment]
@junrushao1994 @comaniac @areusch I just added the scripts and cleaned some things up, take another pass if you can

File: tests/lint/git-black.sh
Comment (comaniac) [inline tests/lint/git-black.sh:56]: Seems like `BLACK` is only used for displaying the version. If so, how about move this line to the if-block so that we can get rid of `BLACK` at all.

**Review by comaniac — APPROVED** [review]
LGTM. Just some nits. Thanks!

**Comment by jroesch** [comment]
@tqchen recommended that we first format the entire code base using these settings then try to land the CI parts, going to open a second PR with the fully formatted repo.

**Comment by jroesch** [comment]
This is blocked on #6448 and #6451, once we land those two it should be possible to add the checking to the CI, format once more and land this.

File: tests/lint/git-black.sh
Comment (leandron) [inline tests/lint/git-black.sh:43]: Is this temp file `/tmp/$$.black-format.txt` supposed to be created by redirecting the output of command in line 67?

File: python/tvm/relay/frontend/mxnet_qnn_op_utils.py
Comment (areusch) [inline python/tvm/relay/frontend/mxnet_qnn_op_utils.py:281]: what's up with this line?

File: tests/lint/git-black.sh
Comment (areusch) [inline tests/lint/git-black.sh:72]: I think you either want --diff and assert on an empty output or --check, I don't know what both do?

File: tests/lint/git-black.sh
Comment (areusch) [inline tests/lint/git-black.sh:36]: right now `-i` does both inplace format and check, do we want that?

**Review by areusch — CHANGES_REQUESTED** [review]

File: tests/lint/git-black.sh
Comment (jroesch) [reply tests/lint/git-black.sh:72]: I think diff and check at same time give that behavior if I understand correctly, check makes it return a -1 status code which should fail this entire script.

File: python/tvm/relay/frontend/mxnet_qnn_op_utils.py
Comment (jroesch) [reply python/tvm/relay/frontend/mxnet_qnn_op_utils.py:281]: These are a bunch of random edits that made the results of black happier.

File: tests/lint/git-black.sh
Comment (jroesch) [reply tests/lint/git-black.sh:43]: I think this is hold over from my copy paste from the clang-format script, we shouldn't need this.

File: pyproject.toml
Comment (jroesch) [reply pyproject.toml:47]: These two files actually trigger an internal error with black and I was going to open up a follow up issue to fix these.

**Comment by jroesch** [comment]
I think I addressed all the comments, and bumped the CI.

**Review by tqchen — APPROVED** [review]

**Review by junrushao1994 — APPROVED** [review]

**Comment by junrushao1994** [comment]
There is some flakiness in the CI. Please retrigger and let's merge it in

**Pull request merged**
